database/migration/src/main/resources/changesets/resync_engine_schema.xml

```diff
@@ -0,0 +1,23 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog objectQuotingStrategy="QUOTE_ALL_OBJECTS"
+                   xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
+
+    <changeSet author="kshakir" id="resync-engine-schema" dbms="hsqldb,mysql,postgresql">
+        <modifyDataType tableName="WORKFLOW_STORE_ENTRY" columnName="HEARTBEAT_TIMESTAMP" newDataType="DATETIME(6)"/>
+        <modifyDataType tableName="WORKFLOW_STORE_ENTRY" columnName="SUBMISSION_TIME" newDataType="DATETIME(6)"/>
+        <addNotNullConstraint
+                tableName="WORKFLOW_STORE_ENTRY"
+                columnName="SUBMISSION_TIME"
+                columnDataType="DATETIME(6)"/>
+    </changeSet>
+
+    <changeSet author="kshakir" id="restore_auto_increment_call_caching_hash_entry_id" dbms="postgresql">
+        <modifyDataType
+                tableName="CALL_CACHING_HASH_ENTRY"
+                columnName="CALL_CACHING_HASH_ENTRY_ID"
+                newDataType="BIGINT"/>
```

Comment:
Not `BIGSERIAL`?

Reply:
Thanks for the second set of eyes. It's possible in the case of `modifyDataType` that liquibase would just translate `BIGSERIAL` to `BIGINT`. From what I've read one cannot actually modify from `SERIAL` to `BIGSERIAL`, as `BIGSERIAL` is just an alias for `BIGINT` + default + a sequence.

- https://www.postgresql.org/docs/11/datatype-numeric.html#DATATYPE-SERIAL
- https://stackoverflow.com/questions/52195303/postgresql-primary-key-id-datatype-from-serial-to-bigserial#answer-52195920

After reading through those links I noticed I had forgotten to widen the sequence. In the latest (force) push, I left the `modifyDataType` as `BIGINT` as that's what liquibase actually runs, and put the `alter sequence` fix in too. I also added a test so that others don't repeat the same mistake.